File: backend/app/weekly_preset.py
"""Haftalık ön ayar indeksi (1..15); mobil `weeklyPresetCatalogIndexUtc` ile aynı formül."""

from datetime import datetime, timezone


def weekly_preset_catalog_index(now: datetime | None = None) -> int:
    dt = now or datetime.now(timezone.utc)
    if dt.tzinfo is None:
        dt = dt.replace(tzinfo=timezone.utc)
    weeks = int(dt.timestamp()) // 604800
    return 1 + (weeks % 15)

File: frontend/src/lib/weeklyPreset.ts
/** Original (0) hariç haftalık vitrinde dönen ön ayar sayısı. */
export const WEEKLY_PRESET_COUNT = 15;

const MS_PER_WEEK = 604800000;

/**
 * Haftalık vitrin meta verisi.
 * Backend `weekly_preset_catalog_index` ile aynı: UTC Unix / 604800 s (7 gün).
 */
export function weeklySpotlightMeta(d = new Date()) {
  const weeks = Math.floor(d.getTime() / MS_PER_WEEK);
  const slot = weeks % WEEKLY_PRESET_COUNT;
  return {
    presetIndex: 1 + slot,
    slotOneBased: slot + 1,
    total: WEEKLY_PRESET_COUNT,
  };
}

export function weeklyPresetCatalogIndexUtc(d = new Date()): number {
  return weeklySpotlightMeta(d).presetIndex;
}
